src/circuit/sapling/mod.rs

```diff
@@ -87,6 +94,41 @@ fn expose_value_commitment<E, CS>(
         value_commitment.as_ref().map(|c| c.value)
     )?;
 
+    match enforce {
+        None => {},
+        Some(enforce) => {
+            // Allocate the enforce bit, which is always one if
+            // the value of the note commitment is nonzero.
+            let enforce_bit = boolean::AllocatedBit::alloc(
+                cs.namespace(|| "enforce"),
+                value_commitment.as_ref().map(|c| c.value != 0)
+            )?;
+
+            // Compute the number as a linear combination
+            let mut value_num = num::Num::zero();
+            {
+                let mut coeff = E::Fr::one();
+                for bit in &value_bits {
+                    value_num = value_num.add_bool_with_coeff(CS::one(), bit, coeff);
+
+                    coeff.double();
+                }
+            }
+
+            // value_num * (1 - enforce_bit) = 0
+            // If `value` is zero, `enforce_bit` _can_ be zero.
+            // If `value` is nonzero, `enforce_bit` _must_ be one.
+            cs.enforce(
+                || "enforce validity",
+                |_| value_num.lc(E::Fr::one()),
```

Comment:
Why not ``|lc| lc + value_num.get_variable()``? (This is a question about the API rather than the specific change.)

Reply:
The number isn't representing a variable but a linear combination. `lc` returns a linear combination (with a coefficient to be applied also).